### PR TITLE
feat: Adds Dropout Layer

### LINKED ISSUE #848: Dropout layer
### Is your feature request related to a problem?

Dropout layers are a common way to regularize a neural network.

### Desired solution

* Add a new class `safeds.ml.nn.layers.DropoutLayer`.
* Superclass: `Layer`
* Constructor parameters: 
  * `probability: float`
  * Call `_check_bounds` to ensure that `probability` is between 0 and 1 (both inclusive)
* Implementation:
  * Use PyTorch's [`Dropout`](https://pytorch.org/docs/stable/generated/torch.nn.Dropout.html) internally
  * Pass the probality along and don't work inplace.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by TellemHD**
Closes #848 

### Summary of Changes
- Adds Dropout Layer and UnitTests

### CODE REVIEW

File: src/safeds/ml/nn/layers/_dropout_layer.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, DropoutLayer):
+            return NotImplemented
+        return (self is other) or (self._input_size == other._input_size)
+
+    def __sizeof__(self) -> int:
+        if self._input_size is None:
+            raise ValueError("The input_size is not yet set.")
+        if isinstance(self.input_size, int):
+            return int(self._input_size)
```
Comment (lars-reimann) [inline src/safeds/ml/nn/layers/_dropout_layer.py:98]:
```suggestion
        if isinstance(self.input_size, int):
            return self.input_size
```

Implicit type casting only works for the value you do the `isinstance` check on. Here you've once used `self.input_size` and once `self._input_size` (with underscore).

File: src/safeds/ml/nn/layers/_dropout_layer.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    Raises
+    ------
+    OutOfBoundsError
+        If probability < 0
+        If probability > 1
+    """
+
+    def __init__(self, probability: float):
+        _check_bounds("probability", probability, lower_bound=_OpenBound(0), upper_bound=_OpenBound(1))
+        self.probability = probability
```
Comment (lars-reimann) [inline src/safeds/ml/nn/layers/_dropout_layer.py:33]:
```suggestion
        self._probability = probability
```

Fields should be internal. Add a property method called `probability` to access the value.

File: src/safeds/ml/nn/layers/_internal_layers.py [hunk 3]
```diff
@@ -56,6 +56,18 @@ def forward(self, x: Tensor) -> Tensor:
         return self._fn(self._layer(x))
 
 
+class _InternalDropoutLayer(nn.Module):
+    def __init__(self, propability: float) -> None:
```
Comment (lars-reimann) [inline src/safeds/ml/nn/layers/_internal_layers.py:60]:
```suggestion
    def __init__(self, probability: float) -> None:
```

Use a rename refactoring.

File: src/safeds/ml/nn/layers/_dropout_layer.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+
+class DropoutLayer(Layer):
+    """
+    Create a dropout Layer.
+
+    Parameters
+    ----------
+    probability:
+        The probability of which the input neuron becomes an output neuron
```
Comment (lars-reimann) [inline src/safeds/ml/nn/layers/_dropout_layer.py:22]: It's the opposite of this: This is the probability that an input element is replaced by 0.

File: tests/safeds/ml/nn/test_dropout_workflow.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    ss = StandardScaler(column_names="value")
+    _, train_table = ss.fit_and_transform(train_table)
+    _, test_table = ss.fit_and_transform(test_table)
+    model = NeuralNetworkRegressor(
+        InputConversionTable(),
+        [ForwardLayer(neuron_count=1), DropoutLayer(probability=0.5)],
+    )
+
+    fitted_model = model.fit(train_table.to_tabular_dataset("target"), epoch_size=1, learning_rate=0.01)
+    fitted_model.predict(test_table.remove_columns_except(["value"]))
```
Comment (lars-reimann) [inline tests/safeds/ml/nn/test_dropout_workflow.py:43]: Can be removed.

File: tests/safeds/ml/nn/test_dropout_workflow.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        path=resolve_resource_path(_inflation_path),
+    )
+    table_1 = table_1.remove_columns(["date"])
+    table_2 = table_1.slice_rows(start=0, length=table_1.row_count - 14)
+    table_2 = table_2.add_columns([(table_1.slice_rows(start=14)).get_column("value").rename("target")])
+    train_table, test_table = table_2.split_rows(0.8)
+
+    ss = StandardScaler(column_names="value")
+    _, train_table = ss.fit_and_transform(train_table)
+    _, test_table = ss.fit_and_transform(test_table)
```
Comment (lars-reimann) [inline tests/safeds/ml/nn/test_dropout_workflow.py:36]: Faster to hard code a table. We don't need to test data processing again here.

**Review by lars-reimann — DISMISSED** [review]

**Review by lars-reimann — APPROVED** [review]
